Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -1861,42 +1871,38 @@ void WebProcessPool::processForNavigationInternal(WebPageProxy& page, const API:
+        return { createNewProcess(), nullptr, "Process swap due to Lockdown mode change"_s };
 
     if (processSwapRequestedByClient == ProcessSwapRequestedByClient::Yes)
-        return completionHandler(createNewProcess(), nullptr, "Process swap was requested by the client"_s);
+        return { createNewProcess(), nullptr, "Process swap was requested by the client"_s };
 
     if (!m_configuration->processSwapsOnNavigation())
-        return completionHandler(WTFMove(sourceProcess), nullptr, "Feature is disabled"_s);
+        return { WTFMove(sourceProcess), nullptr, "Feature is disabled"_s };
 
     if (m_automationSession)
-        return completionHandler(WTFMove(sourceProcess), nullptr, "An automation session is active"_s);
+        return { WTFMove(sourceProcess), nullptr, "An automation session is active"_s };
 
     if (!sourceProcess->hasCommittedAnyProvisionalLoads()) {
         tryPrewarmWithDomainInformation(sourceProcess, targetRegistrableDomain);
-        return completionHandler(WTFMove(sourceProcess), nullptr, "Process has not yet committed any provisional loads"_s);
+        return { WTFMove(sourceProcess), nullptr, "Process has not yet committed any provisional loads"_s };
     }
 
     // FIXME: We should support process swap when a window has been opened via window.open() without 'noopener'.
     // The issue is that the opener has a handle to the WindowProxy.
     if (navigation.openedByDOMWithOpener() && !m_configuration->processSwapsOnWindowOpenWithOpener())
-        return completionHandler(WTFMove(sourceProcess), nullptr, "Browsing context been opened by DOM without 'noopener'"_s);
+        return { WTFMove(sourceProcess), nullptr, "Browsing context been opened by DOM without 'noopener'"_s };
 
     // FIXME: We should support process swap when a window has opened other windows via window.open.
     if (navigation.hasOpenedFrames())
-        return completionHandler(WTFMove(sourceProcess), nullptr, "Browsing context has opened other windows"_s);
+        return { WTFMove(sourceProcess), nullptr, "Browsing context has opened other windows"_s };
 
     if (auto* targetItem = navigation.targetItem()) {
         if (auto* suspendedPage = targetItem->suspendedPage()) {
-            return suspendedPage->waitUntilReadyToUnsuspend([createNewProcess = WTFMove(createNewProcess), completionHandler = WTFMove(completionHandler)](SuspendedPageProxy* suspendedPage) mutable {
-                if (!suspendedPage)
-                    return completionHandler(createNewProcess(), nullptr, "Using new process because target back/forward item's suspended page is not reusable"_s);
-                Ref<WebProcessProxy> process = suspendedPage->process();
-                completionHandler(WTFMove(process), suspendedPage, "Using target back/forward item's process and suspended page"_s);
-            });
+            Ref<WebProcessProxy> process = suspendedPage->process();
```

Comment:
I feel like this line can be put into the next line.
<WebProcessProxy> is also probably unneeded.